03week/ticTacToe.js

```diff
@@ -14,34 +15,85 @@ let board = [
 
 let playerTurn = 'X';
 
+//each row on the board is a seperate array. Each of these arrays are set to a
+//different variable to more easily mainpulate them
+// let rowZero = board[0]
+// let rowOne = board[1]
+// let rowTwo = board[2]
+
 function printBoard() {
   console.log('   0  1  2');
   console.log('0 ' + board[0].join(' | '));
   console.log('  ---------');
   console.log('1 ' + board[1].join(' | '));
   console.log('  ---------');
   console.log('2 ' + board[2].join(' | '));
-}
+};
 
 function horizontalWin() {
-  // Your code here
-}
+  //checks for the 3 cases of a horizontal win for each player
+  if ((board[0][0] == 'X' && board[0][1] == 'X' && board[0][2] == 'X') ||
+    (board[1][0] == 'X' && board[1][1] == 'X' && board[1][2] == 'X') ||
+    (board[2][0] == 'X' && board[2][1] == 'X' && board[2][2] == 'X') ||
+    (board[0][0] == 'O' && board[0][1] == 'O' && board[0][2] == 'O') ||
+    (board[1][0] == 'O' && board[1][1] == 'O' && board[1][2] == 'O') ||
+    (board[2][0] == 'O' && board[2][1] == 'O' && board[2][2] == 'O')) {
+    console.log(`${playerTurn} wins!`)
+    return true
+  } else {//if no win found then it exits to verticalWin func
+    return verticalWin();
+  }
+
+};
 
 function verticalWin() {
-  // Your code here
-}
+  //checks for the 3 cases of a vert win for each player
+  if ((board[0][0] === 'X' && board[1][0] === 'X' && board[2][0] === 'X') ||
+    (board[0][1] === 'X' && board[1][1] === 'X' && board[2][1] === 'X') ||
+    (board[0][2] === 'X' && board[1][2] === 'X' && board[2][2] === 'X') ||
+    (board[0][0] === 'O' && board[1][0] === 'O' && board[2][0] === 'O') ||
+    (board[0][1] === 'O' && board[1][1] === 'O' && board[2][1] === 'O') ||
+    (board[0][2] === 'O' && board[1][2] === 'O' && board[2][2] === 'O') ) {
+    console.log(`${playerTurn} wins!`)
+    return true
+  } else {//if not win found then it exits to diagonlWin func
+    return diagonalWin();
+  }
+};
 
 function diagonalWin() {
-  // Your code here
-}
-
-function checkForWin() {
-  // Your code here
-}
+  //checks for last 2 cases of win, diagonally, for each player
+  if ((board[0][0] === 'X' && board[1][1] === 'X' && board[2][2] === 'X') ||
```

Comment:
You're repeating yourself A LOT here. You don't need to check for both X and O, just player turn